extensions/cornerstone-dicom-seg/src/getSopClassHandlerModule.js

```diff
@@ -0,0 +1,312 @@
+  const labelmapVolumeId = segDisplaySet.displaySetInstanceUID;
+  const segVolume = SegmentationService.getLabelmapVolume(labelmapVolumeId);
+
+  return segVolume !== undefined;
+}
+
+function _getPixelData(dataset, segments) {
+  let frameSize = Math.ceil((dataset.Rows * dataset.Columns) / 8);
+  let nextOffset = 0;
+
+  Object.keys(segments).forEach(segmentKey => {
+    const segment = segments[segmentKey];
+    segment.numberOfFrames = segment.functionalGroups.length;
+    segment.size = segment.numberOfFrames * frameSize;
+    segment.offset = nextOffset;
+    nextOffset = segment.offset + segment.size;
+    const packedSegment = dataset.PixelData[0].slice(
+      segment.offset,
+      nextOffset
+    );
+
+    segment.pixelData = dcmjs.data.BitArray.unpack(packedSegment);
+    segment.geometry = geometryFromFunctionalGroups(
+      dataset,
+      segment.functionalGroups
+    );
+  });
+
+  return segments;
+}
+
+function geometryFromFunctionalGroups(dataset, perFrame) {
+  let pixelMeasures =
+    dataset.SharedFunctionalGroupsSequence.PixelMeasuresSequence;
+  let planeOrientation =
+    dataset.SharedFunctionalGroupsSequence.PlaneOrientationSequence;
+  let planePosition = perFrame[0].PlanePositionSequence; // TODO: assume sorted frames!
+
+  const geometry = {};
+
+  // NB: DICOM PixelSpacing is defined as Row then Column,
+  // unlike ImageOrientationPatient
+  let spacingBetweenSlices = pixelMeasures.SpacingBetweenSlices;
+  if (!spacingBetweenSlices) {
+    if (pixelMeasures.SliceThickness) {
+      console.log('Using SliceThickness as SpacingBetweenSlices');
+      spacingBetweenSlices = pixelMeasures.SliceThickness;
+    }
+  }
+  geometry.spacing = [
+    pixelMeasures.PixelSpacing[1],
+    pixelMeasures.PixelSpacing[0],
+    spacingBetweenSlices,
+  ].map(Number);
+
+  geometry.dimensions = [dataset.Columns, dataset.Rows, perFrame.length].map(
+    Number
+  );
+
+  let orientation = planeOrientation.ImageOrientationPatient.map(Number);
+  const columnStepToPatient = orientation.slice(0, 3);
+  const rowStepToPatient = orientation.slice(3, 6);
+  geometry.planeNormal = [];
+  vtkMath.cross(columnStepToPatient, rowStepToPatient, geometry.planeNormal);
+
+  let firstPosition = perFrame[0].PlanePositionSequence.ImagePositionPatient.map(
+    Number
+  );
+  let lastPosition = perFrame[
+    perFrame.length - 1
+  ].PlanePositionSequence.ImagePositionPatient.map(Number);
+  geometry.sliceStep = [];
+  vtkMath.subtract(lastPosition, firstPosition, geometry.sliceStep);
+  vtkMath.normalize(geometry.sliceStep);
+  geometry.direction = columnStepToPatient
+    .concat(rowStepToPatient)
+    .concat(geometry.sliceStep);
+  geometry.origin = planePosition.ImagePositionPatient.map(Number);
+
+  return geometry;
+}
+
+function _getSegments(dataset) {
+  const segments = {};
+
+  dataset.SegmentSequence.forEach(segment => {
+    const cielab = segment.RecommendedDisplayCIELabValue;
```

Comment:
Here can be crashed. RecommendedDisplayCIELabValue is optional in dicom

Reply:
Thanks I will fix it later